Content.Shared/Weapons/Ranged/Components/GunComponent.cs

```diff
@@ -82,6 +81,13 @@ public partial class GunComponent : Component
 
     #endregion
 
+    /// <summary>
+    /// Determines whether the weapon can fire "independently", through various signals or interactions
+    /// that do not involve beings activating the weapon with their hands
+    /// </summary>
+    [DataField]
+    public bool CanShootWithoutUser = false;
+
     /// <summary>
```

Comment:
This should be its own component and subscribe to one of the gun events instead.